internal/pkg/warewulfd/parser.go

```diff
@@ -76,7 +86,11 @@ func parseReq(req *http.Request) (parserInfo, error) {
 		return ret, errors.New("no stage encoded in GET")
 	}
 	if ret.hwaddr == "" {
-		return ret, errors.New("no hwaddr encoded in GET")
+		ret.hwaddr = ArpFind(ret.ipaddr)
```

Comment:
ARP will only work if we're on the same Ethernet subnet as the target host. Is this usually encoded as hwaddr in GET? If not, I'm concerned by any dependence on ARP, as that would restrict Warewulf from being used behind a DHCP relay.